cmd/podman/rm.go

```diff
@@ -80,6 +80,9 @@ func rmCmd(c *cliconfig.RmValues) error {
 			return err
 		}
 		if err != nil {
+			if errors.Cause(err) == libpod.ErrNoSuchCtr {
+				exitCode = 1
```

Comment:
We need to reset `exitCode` if further errors are encountered. eg.:
```
$ podman rm bogus running-container
```

It looks like commit 35432ecaae4a8372 masked this problem for rootless usage, but it still holds for the rootful case. I submitted #2632 for this.